proposals/2180-search-service.md

```diff
@@ -0,0 +1,72 @@
+(see [meeting notes](https://wiki.lfaidata.foundation/pages/viewpage.action?pageId=18481442#MarquezCommunityMeetings&Calendar-Discussiontopics))
+which lead to creating Marquez Issue [#2171](https://github.com/MarquezProject/marquez/issues/2171]).
+Some of missing capabilities include: searching by job facet, SQL content, column names, etc.
+
+Current architecture of Marquez consists of Java rest service with PostgreSQL backend. This may be not sufficient
+to implement rich search feature. On the other hand, the simplicity of current Marquez architecture is extremely practical as fresh install can be easily 
+done in minutes as it only requires database backend to run. 
+
+### Why should we introduce external search backend?
+
+It's worth asking **Why PostgreSQL backend is not enough?**. 
+
+ * Because users want to search almost by anything (any table in our relational database) and being able to do SQL 
+`LIKE` on any column of any Marquez table could be a tedious task which could kill database at the end. 
+ * Within current Marquez architecture it is really important to have service up and running, so that incoming lineage
+events get saved in database. We don't want dummy search query to stop loading the events.
+ * We cannot create TEXT index on all the text fields. Creating index on jsonb columns could be dangerous as we know the 
+size of such fields can be counted in megabytes. 
+
+[ElasticSearch](https://www.elastic.co) is widely used open-source search engine. It is beneficial to have its capabilities in Marquez because:
+ * It is designed to index and search JSON documents and OpenLineage events are JSONs by definition. 
+ * It is capable of searching by selected JSON fields or a whole OpenLineage event.
+ 
+### Pluggable Marquez Search Engine
+
+Some users find it extremely useful to use Marquez as simple UI for browsing OpenLineage events that is really easy
+easy to install. For those ElasticSearch can be an overkill. We still want to have an option to run Marquez without Search
+Service which will mean limited search capabilities but with simple installation steps.
+
+Similarly to database configuration, `yaml` configuration could contain extra section:
+```
+searchService:
+  enabled: true
+  elastic.url: http://my-elastic-cluster:9200
+```
+
+Marquez CLI should be capable of indexing whole `lineage_events` table within ElasticSearch indexes.
+
+Whenever lineage event is received in Marquez, it is sent to ElasticSearch indexes. Keep in mind that assuring uniqueness on
+Elastic side may be difficult, so in case of sending multiple same events Marquez will have a single copy of that (as long
+as it's a single run id), but SearchService will contain duplicate entries. 
+
+New `search` endpoint shall be implemented in Marquez and available only when `searchService` enabled.
+
+### Search Service in action
+
+ * Two indexes in ElasticSearch (ES) need to be created:
+   * `lineage`: to contain whole lineage events,
```

Comment:
Maybe `lineage_events` as `lineage` might indicate that the caller can search the lineage graph?